libqtile/widget/bluetooth.py

```diff
@@ -80,21 +80,21 @@ async def _init_adapter(self):
         bus = await MessageBus(bus_type=BusType.SYSTEM).connect()
         introspect = await bus.introspect(BLUEZ, BLUEZ_PATH)
         obj = bus.get_proxy_object(BLUEZ, BLUEZ_PATH, introspect)
-        iface = obj.get_interface(BLUEZ_PROPERTIES)
```

Comment:
You could actually keep this line (but change `iface` to something else) because you can subscribe to the `PropertiesChanged` signal here rather than using the `add_signal_receiver` method, which is really there for when you're just listening for various signals but not actually connecting to the interface,

```
props = obj.get_interface(BLUEZ_PROPERTIES)
props.on_properties_changed(self._signal_received_device)
```
You'd then need to change the signature for that function:
```
def _signal_received_adapter(self, interface, changed, invalidated):
   ...
```
and look at the changed property to see if it includes what you want.

Doing that means you can lose lines 61-76

Reply:
Very clean! Thanks.